src/pages/[platform]/build-a-backend/add-aws-services/geo/geofences/index.mdx

```diff
@@ -0,0 +1,367 @@
+import "maplibre-gl/dist/maplibre-gl.css";
+
+function Map() {
+  const mapRef = useRef(null); // Reference to the map DOM element
+
+  // Wrapping your code in a useEffect allows us to run initializeMap after the div has been rendered into the DOM
+  useEffect(() => {
+    let map;
+    async function initializeMap() {
+      // You only want to initialize the underlying maplibre map after the div has been rendered
+      if (mapRef.current != null) {
+        map = await createMap({
+          container: mapRef.current,
+          center: [-122.431297, 37.773972],
+          zoom: 11,
+        });
+      }
+
++     const control = new AmplifyGeofenceControl()
++     map.addControl(control);
+  }
+  initializeMap();
+
+    // Cleans up and maplibre DOM elements and other resources - https://maplibre.org/maplibre-gl-js/docs/API/classes/Map/#remove
+    return function cleanup() {
+      if (map != null) map.remove();
+    };
+  }, []);
+
+ return (
+   <div className="App">
+     <div ref={mapRef} id="map" />
+   </div>
+ );
+}
+
+export default withAuthenticator(Map);
+```
+
+  </Block>
+
+  <Block name="React">
+
+**Note:** When using the [Amplify React MapView component](https://ui.docs.amplify.aws/react/components/geo) you can use the [`useControl` hook from react-map-gl](https://visgl.github.io/react-map-gl/docs/api-reference/use-control) to render the Geofence control component. The [react-map-gl](https://visgl.github.io/react-map-gl) dependency is already installed through `@aws-amplify/ui-react-geo`, you do not need to install it manually.
+
+```javascript
+import React from 'react';
+import { Amplify } from 'aws-amplify';
+import { withAuthenticator } from '@aws-amplify/ui-react';
+import { MapView } from '@aws-amplify/ui-react-geo';
+import { useControl } from 'react-map-gl';
+import { AmplifyGeofenceControl } from 'maplibre-gl-js-amplify';
+import '@aws-amplify/ui-react/styles.css';
+import '@aws-amplify/ui-react-geo/styles.css';
+import 'maplibre-gl-js-amplify/dist/public/amplify-ctrl-geofence.css';
+import amplifyconfig from '../amplifyconfiguration.json';
+Amplify.configure(amplifyconfig);
+
+function Geofence() {
+  useControl(() => new AmplifyGeofenceControl());
+
+  return null;
+}
+
+function App({ signOut }) {
+  return (
+    <div className="App">
+      <MapView
+        initialViewState={{
+          latitude: 37.8,
+          longitude: -122.4,
+          zoom: 14
+        }}
+      >
+        <Geofence />
+      </MapView>
+    </div>
+  );
+}
+
+export default withAuthenticator(App);
+```
+
+  </Block>
+
+</BlockSwitcher>
+
+> **Note:** Ensure that your package bundler (webpack, rollup, etc) is configured to handle css files. Check out the webpack documentation [here](https://webpack.js.org/loaders/css-loader/).
+
+![Geofence Create example when selecting an area on a map](/images/geofence-create.png)
+
+![Geofence Edit example when selecting an area on a map](/images/geofence-edit.png)
```

Comment:
Could we get the alt text to be a little bit more descriptive? I think mentioning what the "Create example" and the "Edit example" are supposed to show would be helpful